Check installation and versions of 3rd party binaries

jidicula opened the pull request:
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied the relevant labels to this PR
- [x] I've added relevant tests for my contribution
- [x] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer

<!--- For the title, please observe the following rules:
	- Provide a concise and self-descriptive title
	- Do not include the applicable issue number in the title, do it in the PR body
	- If the PR is not ready for review, convert it to a draft.
-->

## Description
<!-- describe what the PR is about. Explain the approach and possible drawbacks.It's ok to repeat some text from the related issue. -->
ST relies on `prelude` and `dcm2niix` for a lot of its functionality. Therefore, it's necessary to have a simpler tool to check that these dependencies are installed, as well as provide some output about their versions. Additionally, this basic check should also output information about the environment, for debugging purposes.

Some additional refactoring is included in this PR to reduce duplication of functionality in `conftest.py` as a result of #136.

## Linked issues
<!-- If the PR fixes any issues, indicate it here with issue-closing keywords: e.g. Resolves #XX, Fixes #XX, Addresses #XX. Note that if you want multiple issues to be autoclosed on PR merge, you must use the issue-closing verb before each relevant issue: e.g. Resolves #1, Resolves #2 -->
Resolves #161

jidicula committed:
Add CLI to check 3rd-party binary installation

Resolves #161

**Why this change was necessary**
It's more useful to the user to have a separate script that they can
run to check their installation without running the visually
overwhelming testing suite.

**What this change does**
Creates 2 helper functions using the GNU `which` tool to verify that
the `prelude` and `dcm2niic` binaries are both installed and available
in the PATH. Since these helper functions only return the resulting
exit code of the `which` calls, a wrapper function is exposed as a
CLI, with informative text for a human user.

**Any side-effects?**
Note that this duplicates code in conftest.py, which will be
refactored in a subsequent commit to rely on the helper functions
added in this commit.

jidicula committed:
Refactor conftest to use check_env helpers

Related to #161

**Why this change was necessary**
conftest.py had some functionality duplicated in check_env.py, so the
duplicated code was factored out.

**What this change does**
Refactors conftest dependency checks to rely on imported check_env.py.

**Any side-effects?**
conftest now depends on the check_env helper functions for verifying
the prelude and dcm2niix installations.

jidicula committed:
Add helper function to check dcm2niix version

Resolves #161

**Why this change was necessary**
Part of the issue spec is to also check the dependency version if it's
installed.

**What this change does**
Adds function that captures output of `dcm2niix --version` and is
called if `dcm2niix` has been verified to be installed and in the
PATH.

Note that `dcm2niix --version` returns a nonzero exit code. This
helper's approach will be invalid if this behaviour in dcm2niix is
fixed.

**Any side-effects?**
None.

jidicula committed:
Add helper function to check prelude version

Resolves #161

**Why this change was necessary**
Part of the issue spec is to also check the dependency version if it's
installed.

**What this change does**
Adds function that captures output of `prelude --help` and is
called if `prelude` has been verified to be installed and in the PATH.

**Any side-effects?**
Note that `prelude` displays version information as output from the
`--help` flag, but errors out. Therefore, the version must be parsed
from a general help message that's captured from stderr rather than
stdout. This approach will be invalid if/when FSL fixes this
behaviour.

jidicula commented:
TODO

- [x] environment info

- [x] tests

jidicula committed:
Add CLI to dump environment and package info

Resolves #161

**Why this change was necessary**
It's useful to the user to have a CLI tool to dump environment info
into stdout to help ST contributors with debugging.

**What this change does**
Adds 2 helper functions to get environment information including OS
and CPU details, as well as resource usage.

**Any side-effects?**
New dependency is added, so the shimmingtoolbox package will have to
be reinstalled locally.

**Additional context/notes/links**
https://github.com/shimming-toolbox/shimming-toolbox/issues/161#issuecomment-725028653

jidicula committed:
Add tests for check_env module

Resolves #161

**Why this change was necessary**
The newly added environment and soft dependency checker features were
missing tests.

**What this change does**
Adds missing tests to verify functionality of check_env module.

**Any side-effects?**
Increased test coverage.

jidicula committed:
Add 2 new CLI endpoints for check_env module

Resolves #161

**Why this change was necessary**
CLI endpoints should be accessible with the canonical `st_` prefix.

**What this change does**
Adds CLI endpoints for check_env module functions
`check_dependencies()` and `dump_env_info()`.

**Any side-effects?**
2 new CLI endpoints added to `setup.py`.

po09i reviewed (commented):
I really like the output format from sct (see this https://github.com/shimming-toolbox/shimming-toolbox/issues/161#issuecomment-725028653). Currently the output format is rather hard to see. Especially if we add other checks to the cli. (Python packages, ...)

jidicula committed:
Prettify dependency check output

Resolves https://github.com/shimming-toolbox/shimming-toolbox/pull/176#pullrequestreview-530922591
Related to #161

**Why this change was necessary**
A prettier output is more user-friendly and assists with readability[1].

**What this change does**
Uses snippets from SCT to yield prettier output[2] for dependency checks.

**Additional context/notes/links**
[1] https://github.com/shimming-toolbox/shimming-toolbox/issues/161#issuecomment-725028653
[2] https://github.com/neuropoly/spinalcordtoolbox/blob/master/scripts/sct_check_dependencies.py

jcohenadad reviewed (approved):
working nicely 👍 
~~~
(venv) julien-macbook:~/code/shimming-toolbox/shimming-toolbox $ st_check_dependencies
Check if prelude is installed.......................[OK]
    Part of FSL (ID: 6.0.1)
    prelude (Version 2.0 in c# minor)
    Phase Region Expanding Labeller for Unwrapping Discrete Estimates
    Copyright(c) 2001, University of Oxford (Mark Jenkinson)
Check if dcm2niix is installed......................[OK]
    Compression will be faster with 'pigz' installed http://macappstore.org/pigz/
    Chris Rorden's dcm2niiX version v1.0.20200331  (JP2:OpenJPEG) (JP-LS:CharLS) Clang8.1.0 (64-bit MacOS)
    v1.0.20200331
~~~
